File: backend/lib/models/relatorio.dart
import 'funcionario.dart';

enum TipoRelatorio { DOACOES_RECEBIDAS, DOACOES_DISTRIBUIDAS, DOACOES_TODAS }

class Relatorio {
  final int id;
  DateTime dataRelatorio;
  String formato;
  TipoRelatorio tipoRelatorio;
  String? descricao;
  Funcionario funcionario;

  Relatorio({
    required this.id,
    required this.dataRelatorio,
    required this.formato,
    required this.tipoRelatorio,
    this.descricao,
    required this.funcionario,
  });
}

File: backend/app/Models/Relatorio.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Relatorio extends Model
{
    protected $table = 'relatorio';
    public $timestamps = false;
    protected $fillable = [
        'data_relatorio',
        'formato',
        'tipo_relatorio',
        'descricao',
        'funcionario_id'
    ];

    public function funcionario()
    {
        return $this->belongsTo(Funcionario::class, 'funcionario_id');
    }
}
